src/text/text_editor.py

```diff
@@ -46,16 +50,50 @@ class ApplicationState:
     instantiate this as an object and pass at around.
     """
 
-    show_status_bar = True
-    current_path = None
+    def __init__(self):
+        try:
+            with open("user_setting.json", "r") as j:
+                self.user_settings = json.loads(j.read())
+        except FileNotFoundError:
+            # if for some reason the file is deleted but then i need to maintain all the setting here
+            self.user_settings = {"last_path": None, "style": ""}  # color picker?
+            with open("user_setting.json", "w") as j:
+                default_user_settings = json.dumps(self.user_settings)
+                j.write(default_user_settings)
+
+        self.show_status_bar = True
+        if "last_path" in self.user_settings and self.user_settings["last_path"]:
+            self.current_path = self.user_settings["last_path"]
+        else:
+            self.current_path = None
+
+
+appState = ApplicationState()
```

Comment:
We need to move this to a different file. We can't have global variables like this all over the place.

Reply:
Also, should be in snake_case, not camelCase